pkgs/misc/uboot/default.nix

```diff
@@ -65,6 +69,16 @@ in rec {
 
   ubootTools = buildUBoot rec {
     defconfig = "allnoconfig";
+    # This is necessary otherwise the build fails with undefined symbols at link-time.
```

Comment:
At least `ubootTools` builds on 2017.11 without this... Maybe it's been fixed.

Reply:
Just let me know if you have an easy/simple reproducer of this for upstream U-Boot, I have fairly good knowledge of the codebase.

Reply:
The failure is of the form,
```
tools/image-host.o: In function `fit_image_setup_sig':
image-host.c:(.text+0x48b): undefined reference to `image_get_checksum_algo'
image-host.c:(.text+0x4a2): undefined reference to `image_get_crypto_algo'
tools/image-host.o: In function `fit_config_get_data':
image-host.c:(.text+0xf13): undefined reference to `fit_region_make_list'
tools/image-host.o: In function `fit_image_setup_sig':
image-host.c:(.text+0x48b): undefined reference to `image_get_checksum_algo'
image-host.c:(.text+0x4a2): undefined reference to `image_get_crypto_algo'
tools/image-host.o: In function `fit_config_get_data':
image-host.c:(.text+0xf13): undefined reference to `fit_region_make_list'
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
```
Cloning https://github.com/bgamari/nix-cross-test, removing this hunk from the `nixpkgs` submodule, and then running `nix build -f nixos.nix pkgs.ubootTools` is sufficient to reproduce.